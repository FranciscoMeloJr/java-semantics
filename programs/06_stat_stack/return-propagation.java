// Testing that return statement propagates through try-catch blocks.

public class main {

  void main(string[] args) {
    test();
    print("Done!", "\n");
  }

  void test() {
    try {
      try {
        return;

        print("No exception","\n");
      } catch (int e3) {
        print("caught Exception: ",e3,"\n");
      } finally {
        print("finally after return","\n");
      }
    } finally {
      print("finally 2 after return","\n");
    }
  }
}

// finally after return
// finally 2 after return
// Done!
