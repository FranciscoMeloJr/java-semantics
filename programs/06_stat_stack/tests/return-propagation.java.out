finally after return
finally 2 after return
Done!
